### ISSUE TITLE
Optional<T> has incorrect behavior when value for input argument does not exist in gql variables

### ISSUE THREAD

**Issue by tatarenkoigor** [ChilliCream/graphql-platform#6425]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Product

Hot Chocolate

### Describe the bug

When using the Optional<T> type for input arguments, get incorrect behavior when there are no variable values ​​for optional arguments in the gql query. 

```
mutation UpdateCource($id: UUID!, $name: String){
  updateCourse(input: {id:$id, name:$name}){
    id
  }
}

//variables
{
  "id":"08db8c2a-3775-ab62-ee9c-00d956620000"
}
```

I receive in resolver value for this argument {hasValue: true, value: null} 
<img width="407" alt="image" src="https://github.com/ChilliCream/graphql-platform/assets/2392265/33de97e6-53f4-40cc-a181-e2622557515c">

I expect {hasValue: false, value: null(unspecified)}.
<img width="416" alt="image" src="https://github.com/ChilliCream/graphql-platform/assets/2392265/35bbd3cd-7b92-4c02-9329-96709b311e39">

### Steps to reproduce

1. Create input type
```
public class UpdateCource
{
    public Guid Id { get; set; }
    public Optional<string?> Name { get; set; }
}
```

2. Run mutation
```
mutation UpdateCource($id: UUID!, $name: String){
  updateCourse(input: {id:$id, name:$name}){
    id
  }
}

//variables
{
  "id":"08db8c2a-3775-ab62-ee9c-00d956620000"
}
```
3. Wrong result in Name argument
<img width="407" alt="image" src="https://github.com/ChilliCream/graphql-platform/assets/2392265/33de97e6-53f4-40cc-a181-e2622557515c">

4. Expected result in Name argument
<img width="416" alt="image" src="https://github.com/ChilliCream/graphql-platform/assets/2392265/35bbd3cd-7b92-4c02-9329-96709b311e39">

### Version

13.4.0

**Comment by Boooober** [comment]
Really struggling with this stuff on our project.  
Workaround is to write different queries for setting and clearing fields, but this seems like overkill on the frontend side.

**Comment by michaelstaib** [comment]
This is fixed with 13.6.0-rc.1
